src/query.rs

```diff
@@ -280,40 +255,52 @@ mod tests {
             }
         }
     }
+}
+
+macro_rules! add_lints {
+    ($($name:ident,)*) => {
+        $(
+            #[test]
+            fn $name() {
+                tests::check_query_execution(stringify!($name))
+            }
```

Comment:
Doesn't this put this test function outside of `mod tests` and into the regular module that always gets compiled, even in non-test builds?

I like the general idea of having one place to add lints, but I don't think we want to add test-only code that needs to be compiled in non-test builds too.

Reply:
Oh ok, I didn't know what the `#[cfg(test)]` does.

But then, I don't know how to write this code in the `#[cfg(test)] mod tests` module. Rust doesn't allow redefining/extending a `mod`.

Reply:
I'm not sure either. It might be worth looking at other bigger and more established crates (e.g. clippy) to see if they have any tricks that might help here as well.